website/docusaurus.config.js

```diff
@@ -80,6 +85,54 @@ toolsMD.forEach(tool => {
   tool.usage = loadMD(`${tool.path}/usage.md`);
 });
 
+/**
+ * A remark plugin that renders markdown contents within `:::babel8` and the nearest matching `:::` based on the
+ * BABEL_8_BREAKING option. When `BABEL_8_BREAKING` is `true`, contents within `:::babel7` and `:::` will be removed,
+ * otherwise everything within `:::babel8` and `:::` is removed.
+ *
+ * Limit: there must be an empty line before and after `:::babel[78]` and `:::`.
+ *
+ * With this plugin we can maintain both Babel 7 and Babel 8 docs in the same branch.
+ * @param {{BABEL_8_BREAKING: boolean}} options
+ * @returns md-ast transformer
+ */
+function remarkDirectiveBabel8Plugin({ renderBabel8 }) {
+  return async function transformer(root) {
+    visit(root, "paragraph", (node, index, parent) => {
+      const directiveLabel = node.children?.[0].value;
+      if (directiveLabel === ":::babel8" || directiveLabel === ":::babel7") {
+        const siblings = parent.children;
+        let containerEnd = index + 1,
+          nestedLevel = 1;
+        for (; containerEnd < siblings.length; containerEnd++) {
+          const node = siblings[containerEnd];
+          if (node.type === "paragraph") {
+            const directiveLabel = node.children?.[0].value;
+            if (directiveLabel?.startsWith(":::")) {
+              if (directiveLabel.length === 3) {
+                nestedLevel--;
+                if (nestedLevel === 0) {
+                  break;
+                }
+              } else {
+                nestedLevel++;
+              }
+            }
+          }
+        }
+        if (nestedLevel === 0) {
+          if ((directiveLabel === ":::babel8") ^ renderBabel8) {
+            siblings.splice(index, containerEnd - index + 1); // remove anything between ":::babel[78]" and ":::"
```

Comment:
When sibilings are manipulated, the change is not synced to the visit queue, we can see there are still unstripped `:::babel7` directives in https://github.com/babel/website/pull/2776

<img width="468" alt="image" src="https://user-images.githubusercontent.com/3607926/235956640-ac6aa655-a8ee-404f-8334-028d433612f1.png">

I think I have to manipulate the `children` before they are visited.